scripts/variantstore/wdl/ImportGenomes.wdl

```diff
@@ -525,9 +527,55 @@ task LoadTable {
     fi
 
     if [ $NUM_FILES -gt 0 ]; then
-        bq load --location=US --project_id=~{project_id} --skip_leading_rows=1 --source_format=CSV -F "\t" $TABLE $DIR$FILES ~{schema} || exit 1
-        echo "ingested ${FILES} file from $DIR into table $TABLE"
-        gsutil -m mv $DIR$FILES ${DIR}done/
+        # get list of of pet files and their byte sizes
+        echo "Getting load file sizes(bytes) and path to each file."
+        gsutil du "${DIR}${FILES}" | tr " " "\t" | tr -s "\t" > ~{datatype}_du.txt
+
+        # get total memory in bytes
+        echo "Calculating total files' size(bytes)."
+        TOTAL_FILE_SIZE=$(awk '{print $1}' OFS="\t" ~{datatype}_du.txt| paste -sd+ - | bc)
+
+        # get number of iterations to loop through file - round up to get full set of files
+        num_sets=$(((TOTAL_FILE_SIZE+16492674416639)/16492674416640))
+
+        echo "Starting creation of $num_sets set(s) of load files totaling $TOTAL_FILE_SIZE bytes."
+        for set in $(seq 1 $num_sets)
+        do
+          # write set of data totaling 16000000000000 bytes to file labeled by set #
+          awk '{s+=$1}{print $1"\t"$2"\t"s}' ~{datatype}_du.txt | awk '$3 < 16000000000000 {print $1"\t"$2}' > "${set}"_files_to_load.txt
```

Comment:
Thinking out loud here... but maybe this would be easier to do/maintain by creating a file outside of this loop with a column for the "set".  Awk has access to decent math functions, I bet it could be done there.

Then this loop could just grep that file for the set name being loaded
```
awk '{s+=$1}{print $1"\t"$2"\t"s"\t" "set"(1+int(s / 16000000000000))}'~{datatype}_du.txt
```

This adds a new column "setX" where X is the 1-based set of 16gb of files.  You could do this outside the loop and. then the loop could possibly be

```
for set in $(cat file | cut -f4 | sort | uniq)
```

To loop over the unique value of sets.  

Then you. don't have to do this subtracting, temp file stulff, etc

Reply:
brilliant, thats fantastic. implementing that now.